libgalois/src/PropertyGraph.cpp

```diff
@@ -259,8 +259,9 @@ GetEntityTypeIDsFromProperties(
       const std::string& field_name = current_field->name();
       field_names.emplace_back(field_name);
     }
-    katana::EntityTypeID new_entity_type_id =
-        entity_type_manager->AddNonAtomicEntityType(field_names);
+    katana::EntityTypeID new_entity_type_id = KATANA_CHECKED(
+        entity_type_manager->GetOrAddNonAtomicEntityType(KATANA_CHECKED(
```

Comment:
Is this causing failures? Would replacing `GetOrAdd` with `Add` fix the issue?

This should always be `Add`. `type_combinations` is a set, so no duplicates there. Every `i` in `field_names` should have a different name - that should always be the case, otherwise, it's a bug in some prior code.

Reply:
This does add duplicates. It triggered the assert: https://github.com/KatanaGraph/katana/blob/efb0ed41988977cd1dcbfebf19ca31a246785eae/libgalois/include/katana/EntityTypeManager.h#L280-L283

Reply:
Can you assert that the field_name for each index `i` is distinct?

Reply:
The key for the `type_combinations` set is a vector, so two different keys might have the same elements but in different. However, this shouldn't be possible in the code either. Because elements to the key-vector are always inserted in the same order.

Reply:
I'm gonna return to this once https://github.com/KatanaGraph/katana/pull/403 merges, since it might fix or change this behavior anyway. This is very useful information thought. I will come back here and dig in when the time comes.

Reply:
Thanks!